continuous-integration/readme.md

```diff
@@ -1,3 +1,227 @@
+
+## Git Driven Workflow
+
+- [ ] **Build on commit**
+  - Every commit to the baseline repository should trigger the CI pipeline to create a new build candidate.
+  - Build artifact(s) are built, packaged, validated and deployed continuously into a non-production environment per commit. Each commit against the repository results into a CI run which checks out the sources onto the integration machine, initiates a build, and notifies the committer of the result of the build.
+
+- [ ] **Master branch consistency with production**
+  - Merges into the master branch trigger releases into the production environment(s). This way the master branch becomes a dependable baseline for the code running in production.
+
+- [ ] **Avoid commenting out failing tests**
+  - Avoid commenting out tests in the mainline branch. By commenting out tests, we get an incorrect indication of the status of the build.
+
+- [ ] **Branch policy enforcement**
+  - Branch policies should be setup on the master branch so that the build pipeline status becomes a pre-req validation prior to starting a code review. Code review approvers will only start reviewing a pull request once the CI pipeline run passes for the latest pushed git commit.
+  - Broken builds should block pull request reviews.
+
+## Deliver Quickly and Daily
+
+"By committing regularly, every committer can reduce the number of conflicting changes. Checking in a week's worth of work runs the risk of conflicting with other features and can be very difficult to resolve. Early, small conflicts in an area of the system cause team members to communicate about the change they are making."
+
+In the spirit of transparency and embracing frequent communication across a dev crew, we encourage developers to commit code on a daily cadence. This approach provides visibility to feature progress and accelerates pair programming across the team. Here are some principles to consider:
+
+- [ ] **Everyone commits to the git repository each day**
+  - End of day checked-in code should contain unit tests at the minimum.
+  - Run the build locally before checking in to avoid CI pipeline failure saturation. You should verify what caused the error, and try to solve it as soon as possible instead of committing your code. We encourage developers to follow a [lean SDLC principles](https://leankit.com/learn/lean/principles-of-lean-development/).
+  - Isolate work into small chunks which ties directly to business value and refactor incrementally.
+
+## Isolated Environments
+
+One of the key goals of build validation is to isolate and identify failures in non-production environments and minimize any disruption to live production traffic. Our E2E automated tests should run in an environment which mimics our production environment(as much as possible). This includes consistent software versions, OS, test data volume simulations, etc.
```

Comment:
We could mention automated test in production approaches for scenarios where a high fidelity test environments are not feasible.